android/TritonClient/app/src/main/java/com/mobiledgex/tritonclient/MainActivity.java

```diff
@@ -0,0 +1,251 @@
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+import androidx.navigation.ui.AppBarConfiguration;
+import androidx.navigation.ui.NavigationUI;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.navigation.NavigationView;
+import com.google.android.material.snackbar.Snackbar;
+import com.mobiledgex.matchingengine.util.RequestPermissions;
+import com.mobiledgex.tritonclient.ui.home.HomeFragment;
+import com.mobiledgex.tritonlib.InceptionProcessorActivity;
+import com.mobiledgex.tritonlib.InceptionProcessorFragment;
+import com.mobiledgex.tritonlib.Yolov4ProcessorActivity;
+import com.mobiledgex.tritonlib.Yolov4ProcessorFragment;
+import com.mobiledgex.tritonlib.SettingsActivity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
+
+    private static final String TAG = "MainActivity";
+    private AppBarConfiguration mAppBarConfiguration;
+    private RequestPermissions mRpUtil;
+
+    protected static String mHostname;
+    protected static String mCarrierName;
+    protected static String mRegionName;
+    protected static String mAppName;
+    protected static String mAppVersion;
+    protected static String mOrgName;
+
+    public static final int RC_STATS = 2;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        /**
+         * MatchingEngine APIs require special user approved permissions to READ_PHONE_STATE and
+         * one of the following:
+         * ACCESS_FINE_LOCATION or ACCESS_COARSE_LOCATION.
+         *
+         * The example RequestPermissions utility creates a UI dialog, if needed.
+         *
+         * You can do this anywhere, MainApplication.onActivityResumed(), or a subset of permissions
+         * onResume() on each Activity.
+         *
+         * Permissions must exist prior to API usage to avoid SecurityExceptions.
+         */
+        mRpUtil = new RequestPermissions();
+
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        FloatingActionButton fab = findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                        .setAction("Action", null).show();
+            }
+        });
+        fab.setVisibility(View.GONE); //TODO: Figure out if we need this button.
```

Comment:
I wonder if this the floating pink button (Which I can't really tell what it's functionality) since I usually use the location button at the top right to reposition the map to my gps position.
<img width =200 src="https://user-images.githubusercontent.com/24863504/124165374-8ba7ba80-da56-11eb-929f-1b1a3f686443.jpg"/>

Reply:
Yes. It's that button. Automatically added by the new project wizard. In the MobiledgeX SDK Demo app, that button initiates the "full flow" -- register, findCloudlet, verifyLocation.

For TritonClient, I guess I have no use for it, and should remove it.